src/frontend/apps/impress/src/features/docs/doc-search/components/DocSearchModal.tsx

```diff
@@ -0,0 +1,109 @@
+import { InView } from 'react-intersection-observer';
+import { css } from 'styled-components';
+import { useDebouncedCallback } from 'use-debounce';
+
+import { Box } from '@/components';
+import {
+  QuickSearch,
+  QuickSearchData,
+} from '@/components/quick-search/QuickSearch';
+import { QuickSearchGroup } from '@/components/quick-search/QuickSearchGroup';
+import EmptySearchIcon from '@/features/docs/doc-search/assets/illustration-docs-empty.png';
+import { useResponsiveStore } from '@/stores';
+
+import { Doc, useInfiniteDocs } from '../../doc-management';
+
+import { DocSearchItem } from './DocSearchItem';
+
+type DocSearchModalProps = ModalProps & {};
+
+export const DocSearchModal = ({ ...modalProps }: DocSearchModalProps) => {
+  const { t } = useTranslation();
+  const router = useRouter();
+  const [search, setSearch] = useState('');
+  const { isDesktop } = useResponsiveStore();
+  const {
+    data,
+    isFetching,
+    isRefetching,
+    isLoading,
+    fetchNextPage,
+    hasNextPage,
+  } = useInfiniteDocs({
+    page: 1,
+    title: search,
+  });
+  const loading = isFetching || isRefetching || isLoading;
+  const handleInputSearch = useDebouncedCallback(setSearch, 300);
+
+  const handleSelect = (doc: Doc) => {
+    router.push(`/docs/${doc.id}`);
+    modalProps.onClose?.();
+  };
+
+  const docsData: QuickSearchData<Doc> = useMemo(() => {
+    const docs = data?.pages.flatMap((page) => page.results) || [];
+
+    return {
+      groupName: t('Select a document'),
+      elements: search ? docs : [],
+      endActions: hasNextPage
+        ? [{ content: <InView onChange={() => void fetchNextPage()} /> }]
+        : [],
+    };
+  }, [data, hasNextPage, fetchNextPage, t, search]);
+
+  return (
+    <Modal
+      {...modalProps}
+      closeOnClickOutside
+      size={isDesktop ? ModalSize.LARGE : ModalSize.FULL}
+    >
+      <Box
+        aria-label={t('Share modal')}
+        $direction="column"
+        $justify="space-between"
+        $css={css`
+          [cmdk-item] {
+          }
+        `}
+      >
+        <QuickSearch
+          placeholder={t('Type the name of a document')}
+          loading={loading}
+          data={[]}
+          onFilter={handleInputSearch}
+        >
+          <Box $height={isDesktop ? '500px' : 'calc(100vh - 68px - 1rem)'}>
+            {search.length === 0 && (
+              <Box
+                $direction="column"
+                $height="100%"
+                $align="center"
+                $justify="center"
+              >
+                <Image
+                  width={320}
+                  src={EmptySearchIcon}
+                  alt={t('No active search')}
+                />
+              </Box>
+            )}
+            {search && docsData.elements.length > 0 && (
+              <QuickSearchGroup
+                onSelect={handleSelect}
+                group={docsData}
+                renderElement={(doc) => <DocSearchItem doc={doc} />}
+              />
```

Comment:
Could be improved, DocSearchItem rerenders even when docsData is the same. For this usecase it is ok I think.

Reply:
yeah i think it's ok for this use case too